src/test/scala/com/microsoft/hyperspace/index/RefreshIndexTests.scala

```diff
@@ -0,0 +1,245 @@
+          val fileToDelete =
+            if (loc.equals(nonPartitionedDataPath)) deleteDataFile(nonPartitionedDataPath)
+            else deleteDataFile(partitionedDataPath, true)
+
+          val originalIndexDF = spark.read.parquet(s"$systemPath/${indexConfig.indexName}/" +
+            s"${IndexConstants.INDEX_VERSION_DIRECTORY_PREFIX}=0")
+          val delCount =
+            originalIndexDF
+              .filter(s"""${IndexConstants.DATA_FILE_NAME_COLUMN} ==  "$fileToDelete"""")
+              .count()
+
+          hyperspace.refreshIndex(indexConfig.indexName)
+
+          val refreshedIndexDF = spark.read.parquet(s"$systemPath/${indexConfig.indexName}/" +
+            s"${IndexConstants.INDEX_VERSION_DIRECTORY_PREFIX}=1")
+
+          // Validate only index records whose lineage is the deleted file are removed.
+          assert(refreshedIndexDF.count() == (originalIndexDF.count() - delCount))
+
+          val lineageFileNames = refreshedIndexDF
+            .select(IndexConstants.DATA_FILE_NAME_COLUMN)
+            .distinct()
+            .collect()
+            .map(r => new Path(r.getString(0)))
+
+          assert(!lineageFileNames.contains(fileToDelete))
+        }
+      }
+    }
+  }
+
+  test(
+    "Validate refresh index (to handle deletes from the source data)" +
+      "fails as expected on an index without lineage.") {
+    SampleData.save(
+      spark,
+      nonPartitionedDataPath,
+      Seq("Date", "RGUID", "Query", "imprs", "clicks"))
+    val nonPartitionedDataDF = spark.read.parquet(nonPartitionedDataPath)
+
+    withSQLConf(
+      IndexConstants.INDEX_LINEAGE_ENABLED -> "false",
+      IndexConstants.REFRESH_DELETE_ENABLED -> "true") {
+      hyperspace.createIndex(nonPartitionedDataDF, indexConfig)
+
+      deleteDataFile(nonPartitionedDataPath)
+
+      val ex = intercept[HyperspaceException](hyperspace.refreshIndex(indexConfig.indexName))
+      assert(
+        ex.getMessage.contains(s"Index refresh (to handle deleted source data) is" +
+          " only supported on an index with lineage."))
+    }
+  }
+
+  test(
+    "Validate refresh index (to handle deletes from the source data)" +
+      " is aborted if no source data file is deleted.") {
+    SampleData.save(
+      spark,
+      nonPartitionedDataPath,
+      Seq("Date", "RGUID", "Query", "imprs", "clicks"))
+    val nonPartitionedDataDF = spark.read.parquet(nonPartitionedDataPath)
+
+    withSQLConf(
+      IndexConstants.INDEX_LINEAGE_ENABLED -> "true",
+      IndexConstants.REFRESH_DELETE_ENABLED -> "true") {
+      hyperspace.createIndex(nonPartitionedDataDF, indexConfig)
+
+      val ex = intercept[HyperspaceException](hyperspace.refreshIndex(indexConfig.indexName))
+      assert(ex.getMessage.contains("Refresh aborted as no deleted source data file found."))
+    }
+  }
+
+  test(
+    "Validate refresh index (to handle deletes from the source data)" +
+      " fails as expected when all source data files are deleted.") {
```

Comment:
ditto

Reply:
which other comment this "ditto" is referring to? If it is about above one on index usage with v=0 and v=1, we dont have any index usage here. It is a negative test case checking the failure exception. @imback82  Can you plz be more specific about your suggestion here?

Reply:
```
"Validate refresh index (to handle deletes from the source data) " +
  "fails as expected when all source data files are deleted.") {
``` 
I thought it was pretty obvious...